coveralls/reporter.py

```diff
@@ -58,12 +58,31 @@ def get_hits(self, line_num, analysis):
             return 0
         if line_num not in analysis.statements:
             return None
-        if not analysis.has_arcs():
-            return 1
-        if line_num in analysis.missing_branch_arcs():
-            return 0
         return 1
 
+    def get_arcs(self, analysis):
+        """ Hit stats for each branch.
+
+            Compacted array of:
```

Comment:
Just some comment clarifications:

- Please clarify that it is the _return_ value of this function that is this compacted array
- Please also remark on branches being an extended list of repeating sub-lists rather than a list-of-lists. See my comment in the test section about the mod-4 thing being a bit confusing.